### PR TITLE
Implement multi-argument min0 and max0 using lfortran_intrinsic

### PR DESCRIPTION
**Pull request by gptsarthak**
This is simple working fix for the min0 and max0 intrinsic functions.
The following now works and produces the correct ASR.
```fortran
program intrinsics_52
integer :: a, b, c, d, e, f
a = 2
b = -3
c = 5
d = 10
e = 20
f = -30

if (max0(10,20) /= 20) error stop
if (max0(3,5,4) /= 5) error stop
if (max0(a,b) /= a) error stop
if (max0(a,b,c) /= c) error stop
if (max0(d,e,f,a,b,c) /= e) error stop

if (min0(10,20) /= 10) error stop
if (min0(3,5,4) /= 3) error stop
if (min0(a,b) /= b) error stop
if (min0(c,b,a) /= b) error stop
if (min0(d,e,f,a,b,c) /= f) error stop

end
```

### CODE REVIEW

**Comment by gptsarthak** [comment]
We still have to figure out how to make the following work:
```fortran
subroutine fppasu()
integer max0
mm = max0(1,2)
nplu = max0(1,2/2,1)
end
```
The problem is that it does not recognize that ```max0``` is an intrinsic function, since it has been defined as an integer.
We have to fix this in ```visit_FuncCallOrArray()```

One option was that if we can identify it as an intrinsic function before resolving it in ```v = current_scope->resolve_symbol(var_name);``` , we can use it. But user-defined functions with the same name would be an issue.

My other attempt was at ignoring the 'resolved' max0 symbol if it is not defined as a function (it is just an integer in this case). But it gave an error in one of the other tests.

File: integration_tests/CMakeLists.txt [hunk 1]
```diff
@@ -430,6 +430,7 @@ RUN(NAME intrinsics_48 LABELS gfortran llvm NOFAST)
 RUN(NAME intrinsics_49 LABELS gfortran llvm)
 RUN(NAME intrinsics_50 LABELS gfortran llvm NOFAST)
 RUN(NAME intrinsics_51 LABELS gfortran llvm NOFAST)
+RUN(NAME intrinsics_52 LABELS gfortran ) # min0, max0
```
Comment (certik) [inline integration_tests/CMakeLists.txt:433]: You have to add `llvm` here to get it tested with LFortran. And add NOFAST if the `--fast` doesn't work with this yet.

Comment (gptsarthak) [reply integration_tests/CMakeLists.txt:433]: Yes I'll do that in the next commit.
@Pranavchiku how should we handle https://github.com/lfortran/lfortran/pull/1799#issuecomment-1585714230?

Comment (Pranavchiku) [reply integration_tests/CMakeLists.txt:433]: I guess you have to do something like https://github.com/lfortran/lfortran/blob/main/src/lfortran/semantics/ast_common_visitor.h#L4277 at https://github.com/lfortran/lfortran/blob/main/src/lfortran/semantics/ast_common_visitor.h#L4310

Comment (certik) [reply integration_tests/CMakeLists.txt:433]: max0 has to be an IntrinsicFunction.

**Comment by gptsarthak** [comment]
@Pranavchiku Sorry, I did not see your commits in my PR and just pushed one. Is there a way to undo them?
Also, I did not really understand what the code in your commits is trying to achieve exactly. My 5-line fix in this commit seems enough to resolve the issue. IMO we just needed to check if the function is intrinsic and use it instead. I tested it with the above code and it produces the correct ASR. 
@certik what do you think?

**Comment by Pranavchiku** [comment]
It is fine, to fix this kindly look at https://github.com/lfortran/lfortran/pull/121#issuecomment-1211988387, thank you @czgdp1807 for documenting it so perfectly !!

**Comment by gptsarthak** [comment]
Thank you so much, that @czgdp1807 perfectly described and solved the issue I was facing. Learnt a lot from this rookie mistake. 
This PR is ready for review

**Review by certik — APPROVED** [review]
I think this looks great! Thank you.

**Pull request merged**